### PR TITLE
feat: bring usbboot into drive scanner

### PR DESCRIPTION
**Pull request by Shou**
We leverage the Etcher usbboot SDK and bring it into drive scanner.

Changelog-Entry: Bring usbboot into drive scanner.
Depends: https://github.com/resin-io/etcher/pull/1686

### CODE REVIEW

**Comment by Shou** [comment]
Is this going the right way in terms of design, or do we want to split the two (drives, USB devices) up more?

**Comment by jviotti** [comment]
@Shou Can you show us a screenshot? There are various on-going discussions about running usbboot on the scan phase and only present block devices, so lets see what we decide on before continuing work on this.

File: lib/gui/modules/drive-scanner.js [hunk 1]
```diff
@@ -21,13 +21,60 @@ const _ = require('lodash')
...
+
+    const typeDrives = _.map(drives, (drive) => {
+      return _.assign({}, drive, {
+        type: 'BLOCK_DEVICE'
+      })
+    })
+
+    return usbboot.scan().then((usbDevices) => {
+      const typeUsbDevices = _.map(usbDevices, (usbDevice) => {
+        const description = `Bus ${usbDevice.busNumber} / Address ${usbDevice.deviceAddress}`
```
Comment (lurch) [inline lib/gui/modules/drive-scanner.js]: If such a field is needed, it'd probably be better if it was added into the `usbDevice` object in @jviotti 's usbboot PR? Much like the `usbDevice.name` field you're also making use of below.

**Comment by Shou** [comment]
@jviotti sure, here's the drive selector.

image

File: lib/gui/modules/drive-scanner.js (hunk 1 above)
Comment (Shou) [inline lib/gui/modules/drive-scanner.js]: Sounds good with me, so it's up to @jviotti and his PR.

**Comment by lurch** [comment]
Is this PR still needed, or has it been superseded by something else?

**Comment by jviotti** [comment]
Yeah, I think it was superseded. I'll close it!

**Comment by Shou** [comment]
Can confirm it was superseded.

**Pull request closed without merging**